plugins/modules/mysql_role.py

```diff
@@ -0,0 +1,784 @@
+            module.fail_json(msg=msg)
+
+    return normalized_users
+
+
+def server_supports_roles(cursor, role_impl):
+    """Check if a server supports roles.
+
+    Roles supported since MySQL 8.0.0 and MariaDB 10.0.5
+
+    Args:
+        cursor (cursor): Cursor object of a database Python connector.
+        role_impl (library): A corresponding library depending on a server type (MySQL or MariaDB)
+            and version. Refer to the get_implementation function.
+
+    Returns:
+        bool: True if the server type supports roles, otherwise returns False
+    """
+    return role_impl.supports_roles(cursor)
+
+
+def get_users(cursor):
+    """Get users.
+
+    Args:
+        cursor (cursor): Cursor object of a database Python connector.
+
+    Returns:
+        list: List of tuples (username, hostname).
+    """
+    cursor.execute('SELECT User, Host FROM mysql.user')
+    return cursor.fetchall()
+
+
+def get_grants(cursor, user, host):
+    """Get grants.
+
+    Args:
+        cursor (cursor): Cursor object of a database Python connector.
+        user (str): User name
+        host (str): Host name
+
+    Returns:
+        list: List of tuples like [(grant1,), (grant2,), ... ].
+    """
+    cursor.execute('SHOW GRANTS FOR %s@%s', (user, host))
+    return cursor.fetchall()
+
+
+class Role():
+    """Class to work with MySQL role objects.
+
+    Args:
+        module (AnsibleModule): Object of the AnsibleModule class.
+        cursor (cursor): Cursor object of a database Python connector.
+        is_mariadb (bool): Is a server MariaDB?
+
+    Attributes:
+        module (AnsibleModule): Object of the AnsibleModule class.
+        cursor (cursor): Cursor object of a database Python connector.
+        name (str): Role's name.
+        is_mariadb (bool): Is a server MariaDB?
+        host (str): Role's host.
+        full_name (str): Role's full name.
+        exists (bool): Indicates if a role exists or not.
+        members (set): Set of current role's members.
+    """
+    def __init__(self, module, cursor, name, is_mariadb):
+        self.module = module
+        self.cursor = cursor
+        self.name = name
+        self.is_mariadb = is_mariadb
+        if not self.is_mariadb:
+            self.host = '%'
+            self.full_name = '`%s`@`%s`' % (self.name, self.host)
+        else:
+            self.host = ''
+            self.full_name = '`%s`' % self.name
+
+        self.exists = self.__role_exists()
+        self.members = set()
+
+        if self.exists:
+            self.members = self.__get_members()
+
+    def __role_exists(self):
+        """Check if a role exists.
+
+        Returns:
+            bool: True if the role exists, False if it does not.
+        """
+        if not self.is_mariadb:
+            query = 'SELECT count(*) FROM mysql.user WHERE user = %s AND host = %s', (self.name, self.host)
+        else:
+            query = "SELECT count(*) FROM mysql.user WHERE user = %s AND host = ''", (self.name)
```

Comment:
What about filtering on the `is_role` column?
```
SELECT count(*) FROM mysql.user WHERE user = %s AND is_role = 'Y'
```

Reply:
Done, thanks!